spec/factories/attached_files.rb

```diff
@@ -10,4 +10,10 @@
   trait :with_file do
     file { fixture_file_upload(Rails.root.join('spec/fixtures/files/sul.svg'), 'image/svg+xml') }
   end
+
+  trait :with_preserved_file do
+    file { fixture_file_upload(Rails.root.join('spec/fixtures/files/sul.svg'), 'image/svg+xml') }
+
+    after(:build, &:transform_blob_to_preservation)
+  end
```

Comment:
If you want to build on the `with_file` factory above, you can also consider:

```suggestion
  trait :with_preserved_file do
	with_file
    after(:build, &:transform_blob_to_preservation)
  end
```

Reply:
Oh, nice! Will do that.